resource-managers/kubernetes/integration-tests/dev/dev-run-integration-tests.sh

```diff
@@ -176,4 +180,4 @@ properties+=(
   -Dlog4j.logger.org.apache.spark=DEBUG
 )
 
-$TEST_ROOT_DIR/build/mvn integration-test -f $TEST_ROOT_DIR/pom.xml -pl resource-managers/kubernetes/integration-tests -am -Pscala-$SCALA_VERSION -P$HADOOP_PROFILE -Pkubernetes -Pkubernetes-integration-tests ${properties[@]}
+$TEST_ROOT_DIR/build/mvn install -f $TEST_ROOT_DIR/pom.xml -pl resource-managers/kubernetes/integration-tests $BUILD_DEPENDENCIES_MVN_FLAG -Pscala-$SCALA_VERSION -P$HADOOP_PROFILE -Pkubernetes -Pkubernetes-integration-tests ${properties[@]}
```

Comment:
The `install` phase is after `integration-test`, see default lifecycle at the [maven doc](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html):


Phase | Description
-- | --
... | ...
package | take the compiled code and package it in its distributable format, such as a JAR.
pre-integration-test | perform actions required before integration tests are executed. This may involve things such as setting up the required environment.
**integration-test** | process and deploy the package if necessary into an environment where integration tests can be run.
post-integration-test | perform actions required after integration tests have been executed. This may including cleaning up the environment.
verify | run any checks to verify the package is valid and meets quality criteria.
**install** | install the package into the local repository, for use as a dependency in other projects locally.
...|...


With `install` the local repo will be updated with the fresh jars and in case of `--skip-building-dependencies` the build will reuse those.

Reply:
Well, I know that we need `install` sometimes, but for this one, I'm a little negative for converting `integration-test` to `install` because it pollutes the local repo while testing and reviewing someone-else's PR. It's really painful for committers to clean up the local repo everytime to avoid any side-effects for the other PRs. Is this really inevitable?

Reply:
I would say this is the orherway around, assuming in both PRs we stick to `install`.
So when we call `install` in the first PR we could be sure the fresh jars will be used (even when the artifact we build referencing something outside of the maven reactor) and when we move to the next PR because of `install` we have again the fresh jars.